### DISCUSSION TITLE
Warning: Cannot update a component (`Page`) while rendering a different component (`StoreProvider`). To locate the bad setState() call inside `StoreProvider`, follow the stack trace as described

### DISCUSSION THREAD

**Discussion by tippillow** [vercel/next.js#46835] [Help]
### Summary

How can i fix it? Take official example https://github.com/vercel/next.js/tree/canary/examples/with-mobx. Create page -> /countries/${country}. There i'm using getStaticProps and getStaticPaths. First navigate to /countries/england then click on link usa. As a result you will get warning: "Warning: Cannot update a component (`Page`) while rendering a different component (`StoreProvider`). To locate the bad setState() call inside `StoreProvider`, follow the stack trace as described in".

[gif](https://im4.ezgif.com/tmp/ezgif-4-f79b89f550.gif)

### Example

https://github.com/tippillow/next-mobx-getStaticPath-getStaticProps

**Comment by icyJoseph** [comment]
Haven't used mobx for more than hello-world, but it seems like, because of the route change, `_app` goes through an expected re-render, and at the same time , when country changes, the Page reads the update, through useStore, and re-renders, either of which leads to a `set-state-as-you-render` issue.
